plugins/@grouparoo/hubspot/src/lib/export/destinationMappingOptions.ts

```diff
@@ -19,33 +25,73 @@ export const destinationMappingOptions: DestinationMappingOptionsMethod = async
       },
     },
     properties: {
-      required: [{ key: "email", type: "email" }],
-      known: contactsProperties
-        .filter((contactProperty) => contactProperty.name !== "email")
-        .sort((a, b) => {
-          if (a.name > b.name) return 1;
-          if (a.name < b.name) return -1;
-          return 0;
-        })
-        .map((contactProperty) => {
-          let important = true;
-          if (contactProperty.name.match(/^hs_/)) important = false;
-          if (contactProperty.name.match(/^hubspot_/)) important = false;
-          if (contactProperty.name.match(/^ip_/)) important = false;
-          if (contactProperty.name.match(/^notes_/)) important = false;
-          if (contactProperty.name.match(/^num/)) important = false;
-          if (contactProperty.name.match(/^first_/)) important = false;
-          if (contactProperty.name.match(/^recent_/)) important = false;
-          if (contactProperty.name.match(/^engagements_/)) important = false;
-
-          return {
-            key: contactProperty.name,
-            type: "any",
-            // type: contactProperty.type
-            important,
-          };
-        }),
+      required: required,
+      known: known,
       allowOptionalFromProperties: false,
     },
   };
 };
+
+const mapTypesFromHubspotToGrouparoo = (fieldKey, sendgridType) => {
+  switch (fieldKey) {
+    case "mobilephone":
+    case "phone":
+      return "phoneNumber";
+    case "date_of_birth":
+    case "start_date":
+      return "date";
+  }
+  const map = {
+    text: "string",
+    textarea: "string",
+    datetime: "date",
+    // date: "date", // TODO: add parser treatment for this type.
+    number: "float",
+  };
+  const grouparooType = map[sendgridType];
+  if (grouparooType) {
+    return grouparooType;
+  }
+  return null;
+};
+
+export const getRequiredFields = (): Array<{
+  key: string;
+  type: DestinationMappingOptionsResponseTypes;
+}> => {
+  return [{ key: "email", type: "email" }];
+};
+
+const isImportant = (key): Boolean => {
+  return importantFields.includes(key);
+};
+
+export const getUserFields = async (
+  client: any,
+  appOptions: SimpleAppOptions
+): Promise<
+  Array<{
+    key: string;
+    type: DestinationMappingOptionsResponseTypes;
+    important?: boolean;
+  }>
+> => {
+  const fields = await client.contactsProperties.getAllContactsProperties();
+  const out = [];
+  for (const field of fields) {
+    if (field["name"] !== "email" && !field["readOnlyValue"]) {
+      const type: DestinationMappingOptionsResponseTypes = mapTypesFromHubspotToGrouparoo(
+        field["name"],
+        field["fieldType"]
```

Comment:
Looking at the API response, it looke like `field["type"]` might be better to use here.